ffmpeg fails to build due to vf_scale_vulkan error

┌ ffmpeg git  ......................................... [Recently updated]
├ Changing options to comply to nonfree...
├ Compiling static FFmpeg...
├ Running configure...
├ Running make...
Likely error (tail of the failed operation logfile):
GZIP    libavfilter/vf_scale_cuda.ptx.gz
BIN2C   libavfilter/vf_scale_cuda.ptx.c
CC      libavfilter/vf_scale_cuda.ptx.o
CC      libavfilter/vf_scale_vulkan.o
src/libavfilter/vf_scale_vulkan.c: In function 'init_filter':
src/libavfilter/vf_scale_vulkan.c:223:11: error: implicit declaration of function 'ff_vk_spirv_init'; did you mean 'ff_vk_shader_init'? [-Wimplicit-function-declaration]
  223 |     spv = ff_vk_spirv_init();
      |           ^~~~~~~~~~~~~~~~
      |           ff_vk_shader_init
make: *** [/build/ffmpeg-git/ffbuild/common.mak:91: libavfilter/vf_scale_vulkan.o] Error 1
make failed. Check E:/MABS/build/ffmpeg-git/build-static-64bit/ab-suite.make.log
This is required for other packages, so this script will exit.
  Creating diagnostics file...

Attach E:\MABS\build\logs.zip to the GitHub issue.
Make sure the suite is up-to-date before reporting an issue. It might've been fixed already.

Try running the build again at a later time.

[logs.zip](https://github.com/user-attachments/files/24626761/logs.zip)

--- a/build/media-suite_compile.sh
+++ b/build/media-suite_compile.sh
@@ -2301,7 +2301,7 @@ fi
 _check=(lib{glslang,OSDependent}.a
         libSPIRV{,-Tools{,-opt,-link,-reduce}}.a glslang/SPIRV/GlslangToSpv.h)
 if { [[ $mpv != n ]] ||
-     { [[ $ffmpeg != no ]] && enabled_any libplacebo libglslang; } } &&
+     { [[ $ffmpeg != no ]] && enabled_any libplacebo libglslang libshaderc vulkan; } } &&
     do_vcs "$SOURCE_REPO_GLSLANG"; then
     do_uninstall libHLSL.a "${_check[@]}"
     sed -i "s|command_output(\['git', 'clone',|command_output(\['git', 'clone', '--filter=tree:0',|" ./update_glslang_sources.py
@@ -2312,7 +2312,7 @@ fi
 
 _check=(shaderc/shaderc.h libshaderc_combined.a)
 if { [[ $mpv != n ]] ||
-     { [[ $ffmpeg != no ]] && enabled libplacebo; } } ||
+     { [[ $ffmpeg != no ]] && enabled_any libplacebo libshaderc; } } ||
      ! mpv_disabled shaderc &&
     do_vcs "$SOURCE_REPO_SHADERC"; then
     do_patch "https://raw.githubusercontent.com/m-ab-s/mabs-patches/master/shaderc/0001-third_party-set-INSTALL-variables-as-cache.patch" am
@@ -2503,6 +2503,10 @@ if [[ $ffmpeg != no ]]; then
             grep_and_sed '__declspec(__dllimport__)' "$MINGW_PREFIX"/include/gmp.h \
                 's|__declspec\(__dllimport__\)||g' "$MINGW_PREFIX"/include/gmp.h
         fi
+
+        enabled vulkan && ! enabled_any libshaderc libglslang && do_addOption --enable-libglslang
+        enabled_all libshaderc libglslang && do_removeOption --enable-libglslang
+        enabled libshaderc && sed -ri 's/(require_pkg_config spirv_library "shaderc) >/\1_combined >/' configure
 
         _patches=$(git rev-list $ff_base_commit.. --count)
         if [[ $_patches -gt 0 ]]; then

